### PR TITLE
[AF-1672]: Fixed JMS connection closed when FS deleted

### PR DESCRIPTION
**Pull request by adrielparedes**
There was a problem when file systems were deleted. It also close connection to AMQ.

@ederign would you mind to review?

### CODE REVIEW

**Comment by adrielparedes** [comment]
jenkins execute full downstream build

**Comment by jhrcek** [comment]
@adrielparedes interesting! I'm not sure I understand the fix.
From your description it seems that JMS connection was closed when it shouldn't.
But from the fix it seems you're doing some extra closing.
Could you please explain?

**Comment by adrielparedes** [comment]
@jhrcek actually the "extra closing" is to ensure that when the connection is going to be closed, if any fsWatchServices is still open it also will be closed.

**Comment by jhrcek** [comment]
Jenkins execute full downstream build

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/ws/JGitFileSystemsEventsManager.java [hunk 1]
```diff
@@ -122,6 +122,10 @@ public void close(String name) {
                              ex);
             }
         }
+    }
```
Comment (ederign) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/ws/JGitFileSystemsEventsManager.java]: @jhrcek the git gots weird, but that is the wrong call:

    public void close(String name) {

        JGitFileSystemWatchServices watchService = fsWatchServices.get(name);

        if (watchService != null) {
            try {
                watchService.close();
            } catch (final Exception ex) {
                LOGGER.error("Can't close watch service [" + toString() + "]",
                             ex);
            }
        }
		
		***** THIS PART HERE was removed
		
        if (jGitEventsBroadcast != null) {
            jGitEventsBroadcast.close();
        }
    }

**Comment by jhrcek** [comment]
@ederign after sifting through all the exceptions we see we were able to distil most of the errors to the following 2 exceptions

1. <details>
  <summary>This exception</summary>
  <pre>
08:37:43,141 ERROR [org.uberfire.java.nio.fs.jgit.util.GitImpl] (pool-19-thread-1) Unexpected exception (1/0).: java.io.FileNotFoundException: /home/gluser/mountGluster/.niogit/system/system.git/objects/0d/3e076e88f7a21fea956164b3cf811f78ddc38c (Stale file handle)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at org.eclipse.jgit.internal.storage.file.ObjectDirectory.getLooseObjectSize(ObjectDirectory.java:585)
	at org.eclipse.jgit.internal.storage.file.ObjectDirectory.getObjectSize(ObjectDirectory.java:509)
	at org.eclipse.jgit.internal.storage.file.WindowCursor.getObjectSize(WindowCursor.java:180)
	at org.uberfire.java.nio.fs.jgit.util.commands.GetPathInfo.execute(GetPathInfo.java:73)
	at org.uberfire.java.nio.fs.jgit.util.GitImpl.lambda$getPathInfo$3(GitImpl.java:397)
	at org.uberfire.java.nio.fs.jgit.util.GitImpl.retryIfNeeded(GitImpl.java:473)
	at org.uberfire.java.nio.fs.jgit.util.GitImpl.getPathInfo(GitImpl.java:394)
	at org.uberfire.java.nio.fs.jgit.JGitFileSystemProvider.readAttributes(JGitFileSystemProvider.java:2018)
	at org.uberfire.java.nio.file.Files.readAttributes(Files.java:638)
	at org.uberfire.java.nio.file.Files.size(Files.java:825)
	at org.uberfire.java.nio.file.Files.readAllBytes(Files.java:1186)
	at org.uberfire.io.impl.AbstractIOService.readAllString(AbstractIOService.java:471)
	at org.uberfire.io.impl.AbstractIOService.readAllString(AbstractIOService.java:482)
	at org.guvnor.structure.backend.config.ConfigurationServiceImpl.getConfiguration(ConfigurationServiceImpl.java:300)
	at org.guvnor.structure.backend.config.ConfigurationServiceImpl.getConfiguration(ConfigurationServiceImpl.java:213)
	at org.guvnor.structure.backend.config.ConfigurationServiceImpl$Proxy$_$$_WeldClientProxy.getConfiguration(Unknown Source)
	at org.guvnor.structure.backend.organizationalunit.OrganizationalUnitServiceImpl.loadOrganizationalUnits(OrganizationalUnitServiceImpl.java:121)
	at org.guvnor.structure.backend.organizationalunit.OrganizationalUnitServiceImpl.updateRegisteredOUCache(OrganizationalUnitServiceImpl.java:500)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:95)
	at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:85)
	at org.jboss.weld.injection.MethodInvocationStrategy$SimpleMethodInvocationStrategy.invoke(MethodInvocationStrategy.java:129)
	at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:330)
	at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:308)
	at org.jboss.weld.event.ObserverMethodImpl.notify(ObserverMethodImpl.java:286)
	at javax.enterprise.inject.spi.ObserverMethod.notify(ObserverMethod.java:124)
	at org.jboss.weld.util.Observers.notify(Observers.java:166)
	at org.jboss.weld.event.ObserverNotifier.notifySyncObservers(ObserverNotifier.java:285)
	at org.jboss.weld.event.ObserverNotifier.notify(ObserverNotifier.java:273)
	at org.jboss.weld.event.EventImpl.fire(EventImpl.java:96)
	at org.guvnor.structure.backend.config.watch.ConfigServiceWatchServiceExecutorImpl.execute(ConfigServiceWatchServiceExecutorImpl.java:78)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.as.ee.component.ManagedReferenceMethodInterceptor.processInvocation(ManagedReferenceMethodInterceptor.java:52)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.invocation.InterceptorContext$Invocation.proceed(InterceptorContext.java:509)
	at org.jboss.as.weld.interceptors.Jsr299BindingsInterceptor.doMethodInterception(Jsr299BindingsInterceptor.java:90)
	at org.jboss.as.weld.interceptors.Jsr299BindingsInterceptor.processInvocation(Jsr299BindingsInterceptor.java:101)
	at org.jboss.as.ee.component.interceptors.UserInterceptorFactory$1.processInvocation(UserInterceptorFactory.java:63)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ejb3.component.invocationmetrics.ExecutionTimeInterceptor.processInvocation(ExecutionTimeInterceptor.java:43)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ee.concurrent.ConcurrentContextInterceptor.processInvocation(ConcurrentContextInterceptor.java:45)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.invocation.InitialInterceptor.processInvocation(InitialInterceptor.java:40)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.invocation.ChainedInterceptor.processInvocation(ChainedInterceptor.java:53)
	at org.jboss.as.ee.component.interceptors.ComponentDispatcherInterceptor.processInvocation(ComponentDispatcherInterceptor.java:52)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ejb3.component.singleton.SingletonComponentInstanceAssociationInterceptor.processInvocation(SingletonComponentInstanceAssociationInterceptor.java:53)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ejb3.component.interceptors.AdditionalSetupInterceptor.processInvocation(AdditionalSetupInterceptor.java:54)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ejb3.tx.CMTTxInterceptor.invokeInNoTx(CMTTxInterceptor.java:216)
	at org.jboss.as.ejb3.tx.CMTTxInterceptor.notSupported(CMTTxInterceptor.java:345)
	at org.jboss.as.ejb3.tx.CMTTxInterceptor.processInvocation(CMTTxInterceptor.java:142)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.invocation.InterceptorContext$Invocation.proceed(InterceptorContext.java:509)
	at org.jboss.weld.module.ejb.AbstractEJBRequestScopeActivationInterceptor.aroundInvoke(AbstractEJBRequestScopeActivationInterceptor.java:81)
	at org.jboss.as.weld.ejb.EjbRequestScopeActivationInterceptor.processInvocation(EjbRequestScopeActivationInterceptor.java:89)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ejb3.component.interceptors.CurrentInvocationContextInterceptor.processInvocation(CurrentInvocationContextInterceptor.java:41)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ejb3.component.invocationmetrics.WaitTimeInterceptor.processInvocation(WaitTimeInterceptor.java:47)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ejb3.security.SecurityContextInterceptor.processInvocation(SecurityContextInterceptor.java:100)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ejb3.deployment.processors.StartupAwaitInterceptor.processInvocation(StartupAwaitInterceptor.java:22)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.invocation.InterceptorContext$Invocation.proceed(InterceptorContext.java:509)
	at org.jboss.as.ejb3.component.singleton.ContainerManagedConcurrencyInterceptor.processInvocation(ContainerManagedConcurrencyInterceptor.java:106)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ejb3.component.interceptors.ShutDownInterceptorFactory$1.processInvocation(ShutDownInterceptorFactory.java:64)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ejb3.component.interceptors.LoggingInterceptor.processInvocation(LoggingInterceptor.java:67)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ee.component.NamespaceContextInterceptor.processInvocation(NamespaceContextInterceptor.java:50)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.invocation.ContextClassLoaderInterceptor.processInvocation(ContextClassLoaderInterceptor.java:60)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.invocation.InterceptorContext.run(InterceptorContext.java:438)
	at org.wildfly.security.manager.WildFlySecurityManager.doChecked(WildFlySecurityManager.java:619)
	at org.jboss.invocation.AccessCheckingInterceptor.processInvocation(AccessCheckingInterceptor.java:57)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.invocation.ChainedInterceptor.processInvocation(ChainedInterceptor.java:53)
	at org.jboss.as.ee.component.ViewService$View.invoke(ViewService.java:198)
	at org.jboss.as.ee.component.ViewDescription$1.processInvocation(ViewDescription.java:185)
	at org.jboss.as.ee.component.ProxyInvocationHandler.invoke(ProxyInvocationHandler.java:81)
	at org.guvnor.structure.backend.config.watch.ConfigServiceWatchServiceExecutor$$$view2.execute(Unknown Source)
	at org.guvnor.structure.backend.config.ConfigurationServiceImpl$CheckConfigurationUpdates.execute(ConfigurationServiceImpl.java:529)
	at org.guvnor.structure.backend.config.ConfigurationServiceImpl$1.run(ConfigurationServiceImpl.java:152)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)



08:37:43,147 ERROR [org.jboss.as.ejb3.invocation] (pool-19-thread-1) WFLYEJB0034: EJB Invocation failed on component ConfigServiceWatchServiceExecutorImpl for method public abstract void org.guvnor.structure.backend.config.watch.ConfigServiceWatchServiceExecutor.execute(org.uberfire.java.nio.file.WatchKey,long,org.guvnor.structure.backend.config.watch.AsyncWatchServiceCallback): javax.ejb.EJBException: java.lang.RuntimeException: java.io.FileNotFoundException: /home/gluser/mountGluster/.niogit/system/system.git/objects/0d/3e076e88f7a21fea956164b3cf811f78ddc38c (Stale file handle)
	at org.jboss.as.ejb3.tx.CMTTxInterceptor.invokeInNoTx(CMTTxInterceptor.java:223)
	at org.jboss.as.ejb3.tx.CMTTxInterceptor.notSupported(CMTTxInterceptor.java:345)
	at org.jboss.as.ejb3.tx.CMTTxInterceptor.processInvocation(CMTTxInterceptor.java:142)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.invocation.InterceptorContext$Invocation.proceed(InterceptorContext.java:509)
	at org.jboss.weld.module.ejb.AbstractEJBRequestScopeActivationInterceptor.aroundInvoke(AbstractEJBRequestScopeActivationInterceptor.java:81)
	at org.jboss.as.weld.ejb.EjbRequestScopeActivationInterceptor.processInvocation(EjbRequestScopeActivationInterceptor.java:89)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ejb3.component.interceptors.CurrentInvocationContextInterceptor.processInvocation(CurrentInvocationContextInterceptor.java:41)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ejb3.component.invocationmetrics.WaitTimeInterceptor.processInvocation(WaitTimeInterceptor.java:47)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ejb3.security.SecurityContextInterceptor.processInvocation(SecurityContextInterceptor.java:100)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ejb3.deployment.processors.StartupAwaitInterceptor.processInvocation(StartupAwaitInterceptor.java:22)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.invocation.InterceptorContext$Invocation.proceed(InterceptorContext.java:509)
	at org.jboss.as.ejb3.component.singleton.ContainerManagedConcurrencyInterceptor.processInvocation(ContainerManagedConcurrencyInterceptor.java:106)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ejb3.component.interceptors.ShutDownInterceptorFactory$1.processInvocation(ShutDownInterceptorFactory.java:64)
	at org.jboss.invocation.InterceptorContext.proceed(InterceptorContext.java:422)
	at org.jboss.as.ejb3.component.interceptors.LoggingInterceptor.processI..
  </pre>
</details>

Which was happening with **default gluster setup**. I didn't file a jira for that, please let me know if I should do that.

2. After we enabled the gluster parameters recommended in 
```# Necessary to speed up gluster FS lookups
# https://issues.jboss.org/browse/RHDM-71
sudo gluster volume set gv0 features.cache-invalidation on
sudo gluster volume set gv0 performance.nl-cache on



2. NPE when resolving branch []

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/ws/JGitFileSystemsEventsManager.java [hunk 2]
```diff
@@ -122,6 +122,10 @@ public void close(String name) {
...
+
+    public void shutdown() {
+        fsWatchServices.keySet().forEach(jGitFileSystemWatchServices -> this.close(jGitFileSystemWatchServices));
```
[the remark this answers is not in the document]
Comment (adrielparedes) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/ws/JGitFileSystemsEventsManager.java]: @ederign about this, the only I don't really know is what is the relationship between fsManager and fsWatchService. If all FS are closed, all fs watch services are closed too? I tried to find that but I couldn't

[the remark this answers is not in the document]
Comment (adrielparedes) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/ws/JGitFileSystemsEventsManager.java]: @ederign found it, is redundant, deleting and updating

**Pull request merged**